graylog2-server/src/main/java/org/graylog2/grok/GrokPatternRegistry.java

```diff
@@ -83,6 +83,10 @@ public void grokPatternsDeleted(GrokPatternsDeletedEvent event) {
         reload();
     }
 
+    public boolean grokPatternExists(String patternName) {
+        return grokPatternService.loadByName(patternName).isPresent();
```

Comment:
This is hitting the database for every pipeline function call. The `GrokPatternRegistry` is using a cache for caching `Grok` instances. I think we need to add a cache for this method as well. Otherwise this can put too much load on the database.